flake.nix

```diff
@@ -35,11 +36,56 @@
               poetryOverrides
             ];
           };
+          spellcheck = pkgs.writeShellApplication {
+            name = "spellcheck";
+            runtimeInputs = with pkgs; [
+              moreutils
+              ncurses6
+              (hunspellWithDicts (with hunspellDicts; [
+                en-gb-ise
+                en-au
+                en-ca
+                en-us
+              ]))
+            ];
+            text = ''
+              mapfile -t files < <(find . -name '*.md')
+              exceptionsFile=./spelling-exceptions.dic
+              newExceptionsFile=./new-spelling-exceptions.dic
+              gbFile=gb.dic
+              otherFile=other.dic
+              hunspell -d en_GB -l "''${files[@]}" | sed 's/.*: //' | sort | uniq > $gbFile
+              hunspell -d en_US,en_AU,en_CA -l "''${files[@]}" | sed 's/.*: //' | sort | uniq > $otherFile
+              comm -1 -2 $gbFile $otherFile > $newExceptionsFile
+              mapfile -t newExceptions < <(comm -1 -3 $exceptionsFile $newExceptionsFile)
+              cat $exceptionsFile $newExceptionsFile | sort | uniq | sponge $exceptionsFile
+              rm $newExceptionsFile
+              if [ "''${#newExceptions[@]}" -gt 0 ]; then
+                echo New exceptions
+                tput setaf 3
+                IFS=$'\n'; echo "''${newExceptions[*]/#/ - }"
+                tput sgr 0
+                echo
+              fi
```

Comment:
Thanks, that's a good point. The first draft adds the exceptions and shows them to the user, so that they can remove them if they wish; and they would show up in the exceptions file during review, for a second round of scrutiny.

Ideally cspell can make the exceptions file obsolete. If not, I can add some interaction to help guide the user, and of course comments.